docs/api.md

```diff
@@ -644,6 +644,33 @@ most.from([1, 2])
 	.observe(console.log.bind(console));
 ```
 
+### flatMapEnd
+
+####`stream.flatMapEnd(f) -> Stream`
+####`most.flatMapEnd(f, stream) -> Stream`
+
+Map the end event to a new stream, and begin emitting its values.  Note that `f` *must* return a stream.
+
+`function f(x) -> Stream`
+
+```
+stream:            -a-b-c-d-e-f->
+stream.take(4):         -a-b-c-d|
+f(end): 				-1-2-3-4-5->
+stream.flatMapEnd(f): -a-b-c-d-1-2-3-4-5>
+```
```

Comment:
yes i like the 1st diagram with the `f(end)` i dont know if the `x` after the end is needed?